File: week_02/src/main/java/Question10.java
//Autor : Zaid Nayeb
//Date  : 1404/08/26

//Leap Year Checker
//Description: Determine if a given year is a leap year.
//Rule: Year divisible by 4 → leap year, except centuries not divisible by 400.
//Example:
// Input: 2000 Output: Leap year
// Input: 1900 Output: Not leap year


public class Question10 {
    public static String leapYear(int year){
       if ((year % 4 == 0) && (year % 100 != 0 || year % 400 == 0)){
           return "Leap Year";
        }else {
           return "Not Leap Year";
       }

    }
}

File: week_03/src/Question_10.js
// Author : Zaid Nayeb
// Date   : 2025/11/26
/*
10.	Check whether a number is prime 
Description: Determine if a number is divisible only by 1 and itself. 
Example: 
Input: 
17 Output: 
Prime Input: 
20 Output: 
Not Prime 

*/
console.log(isPrime(10))

function isPrime(num) {
  if (num < 2){ 
    return false;
  }
  for (let i = 2; i <= Math.sqrt(num); i++) {
    if (num % i === 0){ 
        return false;
    }

  }
return true;
}
 

module.exports = isPrime;
